openwisp_controller/subnet_division/rule_types/base.py

```diff
@@ -194,8 +194,7 @@ def get_max_subnet(master_subnet, division_rule):
             subnet_obj.full_clean()
             subnet_obj.save()
             max_subnet = subnet_obj.subnet
-        finally:
-            return max_subnet
+        return max_subnet
```

Comment:
Another problem I found here @pandafy was that line 194 above can fail if there's no space for the reserved subnet and when that happens, `max_subnet` is not defined, but `finally` will execute `return max_subnet` anyway (even if line 194 fails), so in that case the code fails because `max_subnet` is not defined.

I think we shouldn't use `finally` here. That statement is designed to be used in situations when it's imperative to call some cleanup operations even in case of unexpected failures.

Reply:
I will keep a note of this.